src/Distribution/Nixpkgs/Haskell/Derivation.hs

```diff
@@ -110,7 +110,7 @@ instance Package Derivation where
 instance NFData Derivation
 
 instance Pretty Derivation where
-  pPrint drv@MkDerivation {..} = funargs (map text ("mkDerivation" : toAscList inputs)) $$ vcat
+  pPrint drv@MkDerivation {..} = funargs (map text ("lib" : "mkDerivation" : toAscList inputs)) $$ vcat
```

Comment:
The `lib` argument does not belong here.

`mkDerivation` is a very special argument, because it's the function that actually constructs a derivation from the given attribute set. We customize the Haskell package by overriding that function to mess with the attribute set in some way, i.e. enabling by enabling shared library builds or disabling test suites, etc. Therefore, it's really important that all Haskell packages take that function as an argument so that we can override it.

`lib`, on the other hand, is not something that we'll ever want to override for a single Haskell derivation, therefore it's not necessary to pass it to every single build individually. Instead, we should pass `lib` (instead of `stdenv`) at the top of `pkgs/development/haskell-modules/hackage-packages.nix` globally for *all* Haskell derivations.

Reply:
> mkDerivation is a very special argument

Yeah, I knew this wasn't the right place: I would have replaced `stdenv` with `lib` but I don't see it appearing in this repository at all. How does it work?

> Instead, we should pass lib (instead of stdenv) at the top of pkgs/development/haskell-modules/hackage-packages.nix globally for all Haskell derivations.

What about standalone expressions?

Reply:
> I would have replaced stdenv with lib but I don't see it appearing in this repository at all. How does it work?

`stdenv` is an argument to the entire package set in `hackage-packages.nix`. The code is generated here: https://github.com/NixOS/cabal2nix/blob/master/hackage2nix/Main.hs#L178

> What about standalone expressions?

Standalone expressions get `stdenv` added to their list of expected arguments by cabal2nix. The code is at: https://github.com/NixOS/cabal2nix/blob/master/src/Cabal2nix.hs#L241

Reply:
Thank you, that helps me at lot! I'm on it.